web/src/message_fetch.ts

```diff
@@ -354,11 +354,6 @@ export function load_messages(opts: MessageFetchOptions, attempt = 1): void {
         // will return an error if appropriate.
         narrow_data = [...narrow_data, ...web_public_narrow];
     }
-    // We don't pass a narrow for the non-spectator, combined feed view; this
-    // is required to display messages if their muted status changes without
-    // a new network request, and so we need the server to send the message
-    // history from muted streams and topics even though the combined feed
-    // view's "in:home" narrow term will filter those.
     if (narrow_data.length > 0) {
         const narrow_param_string = JSON.stringify(narrow_data);
         data.narrow = handle_operators_supporting_id_based_api(narrow_param_string);
```

Comment:
I don't think removing that above block is correct; `public_terms` is `in:home` for that view, which means it will filter muted messages?

![image](https://github.com/user-attachments/assets/b0752deb-0a89-4b74-8e13-1c17a55cff02)

Reply:
The comment that "We don't pass a narrow for the non-spectator, combined feed view;" is inaccurate since we separated `all_messages_data` from combined feed and now we have hooks in place to update combined feed for message update like any other narrow.

So, if set a breakpoint here and reload when narrowed to combined feed, the first fetch will have `narrow_data` empty for the fetch of `all_messages_data` and the second fetch will have `narrow_data` for `in-home` term type.

Reply:
oh! That is true after the next commit df43189c1f57175904dd9d72b52015fa0e6a2216 which is accurate on its own.

Anyways, if we need to pass empty narrow to combined feed for the message updates to work, that's a bug.

Reply:
Hmm, I'm not sure. Can you start a #frontend conversation about how this detail is supposed to work? Deserves some dedicated discussion.

I can merge this version in that this is a rare corner case where this code path is relevant and I'm not sure `main` is correct here either. 

But I think the crux of the issue is that while `in:home` is the correct operator for the combined feed, our current algorithms are to fetch `all_items` for a given view in order to make muting live-update instant... and so the client/server protocol for it can consider skipping that operator in order to send more efficient queries to the server (it's cheaper to not filter muted stuff in terms of database load).